compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1580,7 +1582,16 @@ public void visit(BLangTypeConversionExpr astTypeConversionExpr) {
     @Override
     public void visit(BLangStructLiteral astStructLiteralExpr) {
         List<BIROperand> varDcls = mapToVarDcls(astStructLiteralExpr.enclMapSymbols);
-        visitTypedesc(astStructLiteralExpr.pos, astStructLiteralExpr.getBType(), varDcls);
+        BIROperand anonMap = null;
+        if (astStructLiteralExpr.getBType().getKind() == TypeKind.RECORD) {
+            BRecordTypeSymbol typeSymbol = (BRecordTypeSymbol) astStructLiteralExpr.getBType().tsymbol;
+            if (typeSymbol.annotations != null) {
+                BIRVariableDcl varDcl = this.env.symbolVarMap.get(typeSymbol.annotations);
+                anonMap = new BIROperand(varDcl);
+            }
+        }
```

Comment:
To generate `NewTypeDesc` instruction we need the details of the annotation map.

Reply:
How are we planning to incorporate changes with https://github.com/ballerina-platform/ballerina-lang/pull/39101?

Reply:
Instead of handling in the expression, it should be moved to when visiting the type node.((https://github.com/ballerina-platform/ballerina-lang/blob/ba469c5cdba87c52fd7b2b96331346059cd6159b/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java#L1675))